packages/zowe-explorer/src/job/actions.ts

```diff
@@ -541,3 +542,74 @@ export async function sortJobsBy(jobs: IZoweJobTreeNode, jobsProvider: IZoweTree
     });
     jobsProvider.refresh();
 }
+
+export async function filterJobs(jobsProvider: IZoweTree<IZoweJobTreeNode>): Promise<vscode.InputBox> {
+    let acutal_jobs;
+    let flag = false;
+    for (const level of jobsProvider.mSessionNodes) {
+        if (level.label === "zosmf") {
+            acutal_jobs = level.children;
+            if (level.collapsibleState === 1) {
+                vscode.window.showInformationMessage("Inorder to filter jobs,first populate them using search icon");
+                flag = true;
+            }
+        }
+    }
+    if (flag) return;
+
+    const inputBox = await vscode.window.createInputBox();
+    inputBox.placeholder = "Type here...";
+    inputBox.onDidChangeValue((query) => {
+        query = query.toUpperCase();
+        for (const level of jobsProvider.mSessionNodes) {
+            if (level.label === "zosmf") {
+                level.children = acutal_jobs.filter((item) =>
+                    `${item["job"].jobname as string}(${item["job"].jobid as string}) - ${item["job"].retcode as string}`.includes(query)
+                );
+            }
+        }
+        jobsProvider.refresh();
+    });
+    inputBox.show();
+    return inputBox;
+}
+
+export async function filterSpools(
+    jobsProvider: IZoweTree<IZoweJobTreeNode>,
+    job: IZoweJobTreeNode,
+    zoweFileProvider: IZoweTree<IZoweNodeType>
+): Promise<vscode.InputBox> {
+    if (job["collapsibleState"] == 1) {
+        const spools = await getSpoolFiles(job);
+        const Spools = spools.map((spool) => {
```

Comment:
```suggestion
        const spools = (await getSpoolFiles(job)).map((spool) => {
```

Also, we should catch any errors that may be thrown from `getSpoolFiles` - if an error occurs, pass the error info to the `errorHandling` function